tests/Unit/Elliptic/Systems/Poisson/Solutions/Test_ProductOfSinusoids.cpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <vector>  // IWYU pragma: keep
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/Poisson/Solutions/ProductOfSinusoids.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+SPECTRE_TEST_CASE(
+    "Unit.Elliptic.Poisson.Solutions.ProductOfSinusoids",
+    "[Unit]") {
+  const std::array<double, 3> wave_numbers{{1., 0.5, 3.}};
+  const Poisson::Solutions::ProductOfSinusoids<3> solution(wave_numbers);
+  const tnsr::I<DataVector, 3> positions{
+      {{DataVector{0., 0., M_PI_2, M_PI_2, M_PI, M_PI},
+        DataVector{0., M_PI, 0., M_PI, 0., M_PI},
+        DataVector{size_t{6}, M_PI_2 / 3.}}}};
+
+  CHECK_ITERABLE_APPROX(solution.field(positions).get(),
+                        (DataVector{0., 0., 0., 1., 0., 0.}));
+  CHECK_ITERABLE_APPROX(solution.source(positions).get(),
+                        (DataVector{0., 0., 0., 41. / 4., 0., 0.}));
+
+  test_creation<Poisson::Solutions::ProductOfSinusoids<3>>(
+      "  WaveNumbers: [1, 0.5, 3]");
+}
```

Comment:
test serialization, you could also test random values as is done in tests/Unit/PointwiseFunctions/AnalyticSolutions/NewtonianEulerSolutions/Test_IsentropicVortex.cpp